### PR TITLE
Avoid accessing QgsLayoutItemsListView's model before it is defined

### LINKED ISSUE #39032: Report editor crashes when right clicking on empty Items list
**Describe the bug**
QGIS crashes when right clicking on an empty Items list while editing a report.

**How to Reproduce**
1. Project > New report > give name > ok
2. Right click inside the empty *Items* list on the *Items* panel.
3. Crash

**QGIS and OS versions**
02f5c13ea0 on Debian

### PR DESCRIPTION
**Pull request by uclaros**
## Description
Avoid accessing `QgsLayoutItemsListView`'s model before it is defined

Fixes #39032

### CODE REVIEW

File: src/gui/layout/qgslayoutitemslistview.cpp [hunk 1]
```diff
@@ -115,9 +115,9 @@ void QgsLayoutItemsListView::setCurrentLayout( QgsLayout *layout )
 void QgsLayoutItemsListView::showContextMenu( QPoint point )
 {
   QModelIndex index = indexAt( point );
-  QgsLayoutItem *item = mModel->itemFromIndex( index );
-  if ( !item )
+  if ( !index.isValid() )
     return;
+  QgsLayoutItem *item = mModel->itemFromIndex( index );
```
Comment (elpaso) [inline src/gui/layout/qgslayoutitemslistview.cpp:120]: Why don't you leave the original check for safety?
```c++
  if ( !item )
    return;
```

Comment (uclaros) [reply src/gui/layout/qgslayoutitemslistview.cpp:120]: If `index.isValid()` then `mModel->itemFromIndex( index )` should return an item. I think it's unreachable code, but I can leave it there, sure!

Comment (uclaros) [reply src/gui/layout/qgslayoutitemslistview.cpp:120]: Maybe a single `if ( !mModel ) return;` at the beginning of the method would be better?

Comment (nyalldawson) [reply src/gui/layout/qgslayoutitemslistview.cpp:120]: I agree

**Pull request merged**